pages/index.tsx

```diff
@@ -30,105 +36,18 @@ export default function Home({ featuredPosts, featuredPages, posts, pages }: Hom
-                      </Text>
-                      <Heading as="h2" className="item-title">
-                        <Link
-                          className="global-underline"
-                          href={`${post.slug}`}
-                          textDecoration="none"
-                        >
-                          {post.title}
-                        </Link>
-                      </Heading>
-                      <Text className="item-excerpt">
-                        {post.excerpt}
-                      </Text>
-                      <Box className="global-tags">
-                        {post.tags.map((tag: Tag) => (
-                          <Link
-                            key={tag.id}
-                            textTransform="lowercase"
-                            textDecoration="none"
-                          >#{tag.name}</Link>
-                        ))}
-                      </Box>
-                    </Box>
-                  </Flex>
-                </Box>
-              )
-            } else {
-              return (
-                <Box
-                  as="article"
-                  key={post.id}
-                  className={["item is-image post", index % 2 != 0 ? 'is-even' : 'is-odd',].join(' ')}
-                >
-                  <Flex className="item-container">
-                    <Box className="item-content">
-                      <Link className="item-image global-image" href={`/${post.slug}`}>
-                        <Image
-                          loading="lazy"
-                          src={post.feature_image}
-                          alt="placeholder image" />
-                      </Link>
-                      <Heading as="h2" className="item-title">
-                        <Link className="global-underline" href={`/${post.slug}`} textDecoration="none">{post.title}</Link>
-                      </Heading>
-                      <Text className="global-meta">
-                        {post.primary_author.name}
-                      </Text>
-                      {post.excerpt && <Text className="item-excerpt" fontFamily="one" fontWeight="500" fontSize="13px">
-                        {post.excerpt}
-                      </Text>
-                      }
-                      <Box className="global-tags">
-                        {post.tags.map((tag: Tag) => (
-                          <Link
-                            key={tag.id}
-                            textTransform="lowercase"
-                            textDecoration="none"
-                            href={`/${tag.slug}`}
-                          >#{tag.name}</Link>
-                        ))}
-                      </Box>
-                    </Box>
-                  </Flex>
-                </Box>
-              )
-            }
+            const Article = index === 0 ? MainArticle : SubArticle
+            return <Article key={post.id} post={post} index={index} />
```

Comment:
Awesome. 

There is one more change I'd like. I would like to create one more component called something like `<LoopWrap posts={posts}>` That will take posts in sets of up to five. Something like:

```js
<LoopWrap posts={posts.slice(4)} />
```

Every time a new pagination is loaded it will essentially load another `<LoopWrap posts={posts} />`

This my reference: https://gobankless.ghost.io/
(apologies for duplicate links)

Reply:
@zberwaldt Decided to go with a simpler solution of checking the index but I also added a `Link` component which wraps the `Chakra-UI Link` component with the `NextJS Link` component so we can keep the current styles with the NextJS functionality. Let me know what you think.

Reply:
@jakeacooley 

Lol. Nice. way less complicated than what I had in mind. Can you explain to be the rationale for wrapping the chakra link in the next link?

I don't have issues with the change, just curious about the why.

Reply:
@zberwaldt It comes down to `client-side transitions between routes`. If you look at another branch without the NextJS Link you will notice it white-screens between page transitions and this branch will have a smooth transition.